dialogflow/participant_management_test.py

```diff
@@ -13,20 +13,53 @@
 # limitations under the License.
 
 import os
+from unittest import mock
+
+from google.api_core.operation import Operation
+from google.cloud import dialogflow_v2beta1 as dialogflow
+import pytest
 
 import conversation_management
 import conversation_profile_management
 import document_management
 import knowledge_base_management
 import participant_management
+import test_utils
 
 PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
-KNOWLEDGE_BASE_DISPLAY_NAME = "fake_KNOWLEDGE_BASE_DISPLAY_NAME"
-DOCUMENT_DISPLAY_NAME = "Cancel an order"
+
+CONTENT_URI = "gs://cloud-samples-data/dialogflow/participant_test.html"
 CONVERSATION_PROFILE_DISPLAY_NAME = "fake_conversation_profile"
+DOCUMENT_DISPLAY_NAME = "Cancel an order"
+MIME_TYPE = "text/html"
+KNOWLEDGE_BASE_DISPLAY_NAME = "fake_KNOWLEDGE_BASE_DISPLAY_NAME"
+KNOWLEDGE_BASE_ID = "documents/123"
+KNOWLEDGE_TYPE = "ARTICLE_SUGGESTION"
+
+
+@pytest.fixture(scope="function")
+def mock_create_document_operation():
+    return test_utils.create_mock_create_document_operation(
+        DOCUMENT_DISPLAY_NAME,
+        KNOWLEDGE_BASE_ID,
+        MIME_TYPE,
+        [getattr(dialogflow.Document.KnowledgeType, KNOWLEDGE_TYPE)],
+        CONTENT_URI,
+    )
+
+
+@pytest.fixture(scope="function")
+def mock_document():
+    return test_utils.create_mock_document(
+        DOCUMENT_DISPLAY_NAME,
+        KNOWLEDGE_BASE_ID,
+        MIME_TYPE,
+        [getattr(dialogflow.Document.KnowledgeType, KNOWLEDGE_TYPE)],
+        CONTENT_URI,
+    )
 
 
-def test_analyze_content_text(capsys):
+def test_analyze_content_text(capsys, mock_create_document_operation, mock_document):
```

Comment:
Does mock_create_document_operation need to be a fixture?

Reply:
It doesn't have to be, but it is initializing a shared test resource in another file with some constants. I think it's fine as-is, but please feel free to suggest another approach if needed.

Reply:
Update: I changed it to a constant.